diff --git a/app/service.csirpy b/app/service.csirpy
index 1411066..eef28c5 100644
--- a/app/service.csirpy
+++ b/app/service.csirpy
@@ -1,4 +1,11 @@
 def boot():
-    output = alloc(1)
-    m[output] = 0
-    return output
+    def match(pattern_bytes, records, flags):
+        pattern = pattern_bytes.decode("ascii")
+        records = [record.decode("ascii") for record in records]
+        if flags == "ASCII_IGNORE_CASE":
+            pattern = pattern.lower()
+            records = [record.lower() for record in records]
+        selected = tuple(record.encode("ascii") for record in records if pattern == record)
+        return selected
+    return match
+
