diff --git a/httpx/_models.py b/httpx/_models.py
--- a/httpx/_models.py
+++ b/httpx/_models.py
@@ -1,3 +1,4 @@
+import copy
 import cgi
 import datetime
 import email.message
@@ -181,7 +182,9 @@ class URL:
 
             kwargs["authority"] = authority
 
-        return URL(self._uri_reference.copy_with(**kwargs).unsplit(),)
+        url = copy.copy(self)
+        url._uri_reference = self._uri_reference.copy_with(**kwargs)
+        return url
 
     def join(self, relative_url: URLTypes) -> "URL":
         """
